opentelemetry-exporter-geneva/geneva-uploader/src/client.rs

```diff
@@ -76,10 +80,16 @@ impl GenevaClient {
             cfg.role_name,
             cfg.role_instance,
         );
+        let max_concurrent_uploads = cfg.max_concurrent_uploads.unwrap_or_else(|| {
+            std::thread::available_parallelism()
```

Comment:
We shouldn't use number of cores as the default considering our workload is I/O bound. I think the default concurrency value should match the number of connections in the `reqwest` client's connection pool because we can never execute more HTTP requests in parallel than the available connections.

Reply:
Yes, but I noticed now that there [default connection pool size](https://github.com/seanmonstar/reqwest/blob/5b705ccc6bed87f8bf1b6a980049c6764ffdf493/src/async_impl/client.rs#L289)in `reqwest` client is theoretically unlimited, constraint by the OS limits, or else the available resources. For now, I have added a TODO in the code to find the optimal value:
```rust
        let max_concurrent_uploads = cfg.max_concurrent_uploads.unwrap_or_else(|| {
            // TODO - Use a more sophisticated method to determine concurrency if needed
            // currently using number of CPU cores
            std::thread::available_parallelism()
                .map(|p| p.get())
                .unwrap_or(4)
        });
```